uint16 distance0
uint16 distance1
uint16 distance2
